packages/wallet/src/redux/protocols/funding/readme.md

```diff
@@ -0,0 +1,59 @@
+# Funding Protocol
+
+The purpose of this protocol is to
+
+1. Determine the funding strategy that will be used to fund a channel
+2. Initialize the protocol for the corresponding strategy
+3. Route further actions to that strategy's protocol
+
+It should be triggered by the `FUNDING_REQUESTED` event from the app.
+On success, it should send `FUNDING_SUCCESS` to the app.
+
+Out of scope (for now):
+
+Supporting protocols other than indirect funding
+
+## State machine
+
+The protocol is implemented with the following state machines
+
+### Player A
+
+```mermaid
+graph TD
+  S((start)) --> WFSC(WaitForStrategyChoice)
+
+  WFSC --> |StrategyChosen| WFSR(WaitForStrategyResponse)
+  WFSR --> |StrategyApproved| WFF(WaitForFunding)
+  WFF --> |StrategyAction| WFF
+  WFF --> |StrategyAction| WFPFS(WaitForPostFundSetup)
+
+  WFPFS --> |CommitmentReceived| WFSConf(WaitForSuccessConfirmation)
+  WFSConf --> |ConfirmSuccess| SS((success))
+
+  WFSR --> |StrategyRejected| WFSC
+
+  WFSC --> |Cancel| F((failure))
+  WFSR --> |Cancel| F
+```
+
+### Player B
+
+```mermaid
+graph TD
+  S((start)) --> WFSP(WaitForStrategyProposal)
```

Comment:
I think it makes more sense for player B's wallet to wait for the strategy proposal, then simply ask the app to approve or reject that strategy. If you
1. Wait for funding approval
2. Wait for strategy response
3. Wait for strategy approval

then the user has to approve twice, which is probably confusing.

That approach also adds complexity, since the strategy proposal can arrive before or after the funding approval. My proposed state machine is more linear.